excavator/utils.py

```diff
@@ -78,8 +83,13 @@ def env_int(name, required=False, default=empty):
     environment, a `TypeError` will be raised since we don't want to guess
     about a sensible default.
     """
-    value = get_env_value(name, required=required, default=default)
+    if required and default is not empty:
+        raise ValueError("Using `default` with `required=True` is invalid")
+
+    value = get_env_value(name, required=required, default=empty)
     if value is empty:
+        if default is not empty:
+            return default
         raise ValueError(
```

Comment:
This whole thing is a perfect time for exception handling. Something like the following would make the control flow clearer and cut down on repetition:

```python
def env_int(name, required=False, default=empty):
    """
    Return an integer derived from an environmental variable.  In the case
    where no default is specified and the variable is not present in the
    environment, a `TypeError` will be raised since we don't want to guess
    about a sensible default.
    """
    value = get_env_value(name, required=required, default=default)
    if value is empty:
        raise ValueError(
            "`env_int` requires either a default value to be specified, or for "
            "the variable to be present in the environment"
        )
    try:
        return int(value)
    except TypeError:
        # We have a non-int-able default, let's just return it.
        return value
```